sdformat_urdf/CMakeLists.txt

```diff
@@ -14,18 +14,37 @@ find_package(ament_cmake_ros REQUIRED)
 
 find_package(pluginlib REQUIRED)
 find_package(rcutils REQUIRED)
-find_package(sdformat9 REQUIRED)
 find_package(urdfdom_headers REQUIRED)
 find_package(urdf_parser_plugin REQUIRED)
 find_package(tinyxml2_vendor REQUIRED)
 find_package(TinyXML2 REQUIRED)
 
+# Edifice
+if("$ENV{GAZEBO_VERSION}" STREQUAL "edifice")
+  find_package(sdformat11 REQUIRED)
+  set(SDF_VER ${sdformat11_VERSION_MAJOR})
+
+  message(STATUS "Compiling against Gazebo Edifice")
+# Fortress
+elseif("$ENV{GAZEBO_VERSION}" STREQUAL "fortress")
+  find_package(sdformat12 REQUIRED)
+  set(SDF_VER ${sdformat12_VERSION_MAJOR})
+
+  message(STATUS "Compiling against Gazebo Fortress")
+# Default to libSDFormat 9 for backwards compatibility
```

Comment:
I'd recommend making `citadel` it's own option, and if `GAZEBO_VERSION` isn't specified then default to the latest supported version. My hope is defaulting to the latest would allow this to branch for different ROS distros with less intervention.

```CMake
set(SDF_VER 12)   # Complain about not being able to find this version if no version is found
foreach(major RANGE 12 9)
  find_package(sdformat${major} QUIET)
  if(sdformat${major}_FOUND)
    # Next `find_package` call will be a noop
    set(SDF_VER ${major})
    break()
  endif()
endforeach()
```

Reply:
> if GAZEBO_VERSION isn't specified then default to the latest supported version

This would make Bloom use the latest version, because it doesn't respect custom env vars. 

* https://github.com/ros-infrastructure/bloom/issues/517#issuecomment-821513560

Reply:
Another thing to consider is that the user may have multiple SDF versions installed side-by-side. That's why on `ros_ign` and others we let the user explicitly ask for what version to compile against.

Reply:
aww :(

It looks like bloom supports `ROS_DISTRO`: https://github.com/ros-infrastructure/bloom/pull/519/files#diff-3259a2a5cf932543b3d9745fbe7e3ecf35ba005886ac16cbaa92d703ad31e7e3R133

What about a cascading check? Above these conditionals we check if `GAZEBO_VERSION` is set or not. If it's set, great, but if not then use `ROS_DISTRO` to set the value of `GAZEBO_VERSION`. If neither `GAZEBO_VERSION` nor `ROS_DISTRO` were set then use the latest supported SDFormat version.

If I understand correctly, that should allow bloom to use the appropriate version for a ROS distro, users to pick different versions of Gazebo when building from source, and Rolling to always try building against the latest available.

Reply:
Great idea, done in [34b6c75](https://github.com/ros/sdformat_urdf/pull/8/commits/34b6c75140a79c5c8183098a8e5e8ecf14fc9482), and in [c5afdc9](https://github.com/ros/sdformat_urdf/pull/8/commits/c5afdc9726228b20ca70f799c640d4e173a70027) I renamed `GAZEBO` to `GZ` because that's what we should be using and it completely slipped my mind 🙃